treeherder/webapp/api/performance_data.py

```diff
@@ -825,6 +827,20 @@ def list(self, request):
                 confidence_text, confidence_text_long = perf_compare_utils.get_confidence_text(
                     confidence
                 )
+                sig_hash = (
+                    base_sig.get('signature_hash', '')
+                    if base_sig
+                    else new_sig.get('signature_hash', '')
+                )
+
+                base_push = models.Push.objects.get(
```

Comment:
The logic for `base_push`, `new_push` and `push_timestamp` need to be moved up, out of the for loops, preferably on line `791`. That's the reason why the endpoint seemed to need optimization.

Reply:
```suggestion
            if base_repository_name == new_repository_name:
                repos = [base_repository_name]
            else:
                repos = [base_repository_name, new_repository_name]
            first_push, second_push = models.Push.objects.filter(repository__name=repos,
            revision__in=[base_revision, new_revision])
```
Instead of searching twice through the push objects for base_push and new_push, we can do it once in this way (and we don't actually need to know which push is the base one and which one is the new one).